changelog/_unreleased/2025-01-29-make-rule-classes-internal.md

```diff
@@ -0,0 +1,22 @@
+---
+title: Make Rule classes internal
+issue: NEXT-40440
+author: Jozsef Damokos
+author_email: j.damokos@shopware.com
+author_github: @jozsefdamokos
+---
+# Core
+* Deprecated nearly all rule classes, with the exception of 4, with the intention of making them internal.
+___
+# Upgrade Information
+## Rule classes becoming internal
+* Existing rule classes will be marked as internal, limiting direct usage by third parties.
+* If you currently extend any of the existing rule classes, consider migrating to a custom rule class.
+* Existing rule behavior remains unchanged, but internal implementations may evolve.
+___
+# Next Major Version Changes
+## Rule classes becoming internal
+* Rule classes will become internal, and direct extensions will not be supported.
```

Comment:
```suggestion
* Rule classes are marked internal, and direct extensions is not supported.
```
this will be part of the upgrade guide of the major version, where this actually happend